src/erlcloud_efs.erl

```diff
@@ -0,0 +1,203 @@
+describe_file_systems(AwsConfig) when is_record(AwsConfig, aws_config) ->
+    describe_file_systems(AwsConfig, _Params = []);
+describe_file_systems(Params) ->
+    AwsConfig = erlcloud_aws:default_config(),
+    describe_file_systems(AwsConfig, Params).
+
+-spec describe_file_systems(AwsConfig, Params) -> Result when
+    AwsConfig :: aws_config(),
+    Params :: params(),
+    Result :: {ok, file_systems()} | {ok, file_systems(), token()} | {error, term()}.
+describe_file_systems(AwsConfig, Params) ->
+    Path = [?EFS_API_VERSION, "file-systems"],
+    case request(AwsConfig, _Method = get, Path, Params) of
+        {ok, Response} ->
+            FileSystems = proplists:get_value(<<"FileSystems">>, Response),
+            case proplists:get_value(<<"NextMarker">>, Response, null) of
+                null ->
+                    {ok, FileSystems};
+                Marker ->
+                    {ok, FileSystems, Marker}
+            end;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+-spec describe_file_systems_all() -> Result when
+    Result :: {ok, file_systems()} | {error, term()}.
+describe_file_systems_all() ->
+    AwsConfig = erlcloud_aws:default_config(),
+    describe_file_systems_all(AwsConfig).
+
+-spec describe_file_systems_all(Arg) -> Result when
+    Arg :: aws_config() | params(),
+    Result :: {ok, file_systems()} | {error, term()}.
+describe_file_systems_all(AwsConfig) when is_record(AwsConfig, aws_config) ->
+    describe_file_systems_all(AwsConfig, _Params = []);
+describe_file_systems_all(Params) when is_list(Params) ->
+    AwsConfig = erlcloud_aws:default_config(),
+    describe_file_systems_all(AwsConfig, Params).
+
+-spec describe_file_systems_all(AwsConfig, Params) -> Result when
+    AwsConfig :: aws_config(),
+    Params :: params(),
+    Result :: {ok, file_systems()} | {error, term()}.
+describe_file_systems_all(AwsConfig, Params) ->
+    case describe_file_systems(AwsConfig, Params) of
+        {ok, FileSystems} ->
+            {ok, FileSystems};
+        {ok, FileSystems, Token} ->
+            describe_file_systems_next(AwsConfig, Params, Token, FileSystems);
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+%% -----------------------------------------------------------------------------
+%% Local functions
+%% -----------------------------------------------------------------------------
+
+describe_file_systems_next(AwsConfig, Params, Marker0, FileSystems0) ->
+    case describe_file_systems(AwsConfig, [{<<"Marker">>, Marker0} | Params]) of
+        {ok, FileSystems} ->
+            FileSystems1 = lists:append(FileSystems0, FileSystems),
+            {ok, FileSystems1};
+        {ok, FileSystems, Marker1} ->
+            FileSystems1 = lists:append(FileSystems0, FileSystems),
+            describe_file_systems_next(AwsConfig, Params, Marker1, FileSystems1);
+        {error, Reason} ->
+            {error, Reason}
+    end.
```

Comment:
These appends are not ideiomatic. The Erlangy thing to do would be what erlcloud_ddb_util:flatreverse does. I'm not sure how bad the efficiency impact would actually be realistically with EFS though: https://github.com/erlcloud/erlcloud/blob/ebc0bf4ccc27eff7dddf843404c73723b72180e4/src/erlcloud_ddb_util.erl#L574-L576